src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -179,6 +181,20 @@ public void run() {
                     }
                 }
             });
+        } else if (SET_APPLICATION_ICON_BADGE_NUMBER.equals(action)) {
```

Comment:
No the first method happened only when you called `setApplicationIconBadgeNumber()` directly, right?
I've abstracted it to a public static function in the PushPlugin class, and added it to be called when a `count` it available. Similar to your branch, but when it's `null`, it doesn't change.
Also, you can call it directly (eg. without push, but with `setApplicationIconBadgeNumber()` directly)